### ISSUE TITLE
cp -a from a pipe does not work as Halcyon expects

### ISSUE THREAD

**Issue by mithrandi** [mietek/bashmenot#1]
In 58e4a58913faa28e1d55497db42171a72dba7b74 the `cp` invocation is changed to use `-a` instead of `-p`. However, halcyon invokes `copy_file` using bash's `<(cmd)` syntax, which passes in a pipe as the file argument; this doesn't work with `cp -a`. The actual behaviour is that it just removes the destination entirely, for some reason I don't understand, but the expected behaviour would be to replicate the pipe, which is not desireable either.

I'm not entirely sure if this should be considered a bug in bashmenot, or in halcyon, but the change was in bashmenot so I'm filing it here. Note that this is quite serious, as it breaks setting a large number of options from environment variables or command-line options when using halcyon.

**Comment by mietek** [comment]
Thanks. This will be fixed today. Meanwhile, you can use magic files as an alternative way of setting most options.

**Comment by mietek** [comment]
The reason for changing the `cp` invocation had to do with another corner case, found when using `cp -Rp`.  This was a mistake, and is now reverted.

I have now changed `copy_file` to do a regular non-attribute-preserving `cp`, with the intention of using it whenever only the contents of a file matter, as is the case when setting options from environment variables.

I have also added an attribute-preserving `copy_dir_entries_into` function, intended to be used when the nature of the file itself matters.
